### ISSUE TITLE
className type error 

### ISSUE THREAD

**Issue by j3sch** [marklawlor/tailwindcss-react-native#77]
I created a new project with the expo blank typescript template.
Then I followed the Quick start tutorial to implement tailwindcss-react-native. 

It works perfectly but unfortunately I get an error with typescript error because of className.

`No overload matches this call. #2769
`
`Property 'className' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<View> & Readonly<ViewProps>'. #2772
`

I use VSCode

**Comment by vlaxis** [comment]
Hi @j3sch, create a new `src/tailwindcss-react-native.d.ts` and paste `import "tailwindcss-react-native/types.d";`. See [link from docs](https://tailwindcss-react-native.vercel.app/compilation/babel#typescript)

**Comment by j3sch** [comment]
Perfect, thank you very much

**Comment by binajmen** [comment]
If someone is passing by, last recommendation is here:
https://www.nativewind.dev/getting-started/typescript

**Comment by codymullins** [comment]
@binajmen that worked like a charm

**Comment by ph20Eoow** [comment]
Don't forget to restart your editor. In my case, until I restarted my VScode, the redline was still showing up after adding `app.d.ts`

**Comment by chkilel** [comment]
If you are using _**Neovim**_, what helped is creating a file (eg. `src/nativewind.d.ts`) and pasting this line.

```js
import "nativewind/types"
```

**Comment by marklawlor** [comment]
Using the Triple Slash Directive is the recommended method. Using `import "nativewind/types"` will not work in all scenarios

**Comment by AeRo8** [comment]
In my case, I've created `app.d.ts` and add the below snippet but I still get typescript error because of className. So, i change the file name `app.d.ts` to `app-env.d.ts` and it works.
`/// <reference types="nativewind/types" />`

**Comment by lewisd1996** [comment]
Im still getting this typescript error for some reason: `Property 'className' does not exist on type 'IntrinsicAttributes & ViewProps & RefAttributes<View>'.`

I have a file called `app.d.ts` with `/// <reference types="nativewind/types" />` inside. 

I have tried calling it `my-app.d.ts`, still no luck.

I have restarted TS server, my editor and my machine.

My babel config looks like:

```
module.exports = {
  presets: ['module:metro-react-native-babel-preset'],
  plugins: [
    [
      'module-resolver',
      {
        extensions: ['.ts', '.tsx', '.jsx', '.js', '.json'],
        root: ['.'],
        alias: {
          assets: './src/assets',
          components: './src/components',
          contexts: './src/contexts',
          data: './src/data',
          hooks: './src/hooks',
          navigation: './src/navigation',
          screens: './src/screens',
          providers: './src/providers',
          supabase: './src/supabase',
          services: './src/services',
          config: './src/config',
        },
      },
    ],
    'nativewind/babel',
    'react-native-reanimated/plugin',
  ],
};
```

Anyone see any issues?

**Comment by lewisd1996** [comment]
> Im still getting this typescript error for some reason: `Property 'className' does not exist on type 'IntrinsicAttributes & ViewProps & RefAttributes<View>'.`
> 
> I have a file called `app.d.ts` with `/// <reference types="nativewind/types" />` inside.
> 
> I have tried calling it `my-app.d.ts`, still no luck.
> 
> I have restarted TS server, my editor and my machine.
> 
> My babel config looks like:
> 
> ```
> module.exports = {
>   presets: ['module:metro-react-native-babel-preset'],
>   plugins: [
>     [
>       'module-resolver',
>       {
>         extensions: ['.ts', '.tsx', '.jsx', '.js', '.json'],
>         root: ['.'],
>         alias: {
>           assets: './src/assets',
>           components: './src/components',
>           contexts: './src/contexts',
>           data: './src/data',
>           hooks: './src/hooks',
>           navigation: './src/navigation',
>           screens: './src/screens',
>           providers: './src/providers',
>           supabase: './src/supabase',
>           services: './src/services',
>           config: './src/config',
>         },
>       },
>     ],
>     'nativewind/babel',
>     'react-native-reanimated/plugin',
>   ],
> };
> ```
> 
> Anyone see any issues?

I have just fixed it.

I had to put the `app.d.ts` file into my `src` folder

**Comment by theleanship** [comment]
Had the same issue in a turborepo/yarn workspace setup and the errors would not shift regardless of whichever variation of location + types definition I tried. Eventually realised the workspace hoisted the nativewind module up into the root directory rather than my React Native app. Excluding it from hoisting so it was colocated with the apps node modules finally sorted this out for me.

**Comment by jessemezini** [comment]
Rembember to include app.d.ts to tsconfig.json
```
{
"include": ["app.d.ts"]
}
```

**Comment by dalley8626** [comment]
For some reason i can't get it work at all

tsconfig.json
```
{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "jsx": "react-jsx",
    "module": "esnext",
    "strict": true,
    "baseUrl": "./",
    "paths": {
      "@app/*": ["app/*"],
      "@assets/*": ["assets/*"],
      "@config/*": ["config/*"]
    }
  },
  "tscFilesIncludes": ["app/@types"],
  "include": ["app", "my-app.d.ts", "codegen.ts", "app.config.ts"]
}
```

my-app.d.ts
```
/// <reference types="nativewind/types" />
```

and error is still
```
No overload matches this call.
  Overload 1 of 2, '(props: ViewProps | Readonly<ViewProps>): View', gave the following error.
    Type '{ children: Element[]; className: string; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<View> & Readonly<ViewProps>'.
      Property 'className' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<View> & Readonly<ViewProps>'.
  Overload 2 of 2, '(props: ViewProps, context: any): View', gave the following error.
    Type '{ children: Element[]; className: string; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<View> & Readonly<ViewProps>'.
      Property 'className' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<View> & Readonly<ViewProps>'.ts(2769)
      ```

**Comment by ihassanamjad** [comment]
For those who are still facing this issue, create a new `app.d.ts` file and add a [triple-slash directive](https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html) referencing the types.

`/// <reference types="nativewind/types" />`

Reference: https://www.nativewind.dev/getting-started/typescript

**Comment by dalley8626** [comment]
but now i do have issue with this @ihassanamjad :( 
https://github.com/marklawlor/nativewind/issues/308#issuecomment-1636656928

**Comment by ThatGuySam** [comment]
My team was getting this error because we were passing through all props to a Child Component. 

We were able to resolve the warning when we extended the original Child Components Props using `React.ComponentProps<typeof Svg>` 

```tsx
export type IconProps = {
    name: string
} & React.ComponentProps<typeof Svg>

export default function Icon ( props: IconProps ) {
    const propValues: IconProps = {
        ...props,
        height: props.height || 24,
        width: props.width || 24,
        viewBox: props.viewBox || '0 0 24 24'
    }
    return (
        <View style={ styles.container }>
            <Svg { ...propValues }>
                { getIconPath( props.name ) }
            </Svg>
        </View>
    )
}
```

**Comment by hamashoool** [comment]
> @binajmen that worked like a charm
> 
> tl;dr - create a `my-app.d.ts` and add the below snippet:
> 
> ```
> /// <reference types="nativewind/types" />
> ```

I named mine app.d.ts and it didn't work but, my-app.d.ts worked or any other name

**Comment by holladortun** [comment]
> > @binajmen that worked like a charm
> > tl;dr - create a `my-app.d.ts` and add the below snippet:
> > ```
> > /// <reference types="nativewind/types" />
> > ```
> 
> I named mine app.d.ts and it didn't work but, my-app.d.ts worked or any other name

Thank you very much for this, this worked for me too.

Does anyone know why app.d.ts stopped working? My code worked perfectly using app.d.ts until last week, and now naming the file my-app.d.ts, seems to make it work normally.

**Comment by lavisar** [comment]
> Rembember to include app.d.ts to tsconfig.json
> 
> ```
> {
> "include": ["app.d.ts"]
> }
> ```

This work to me sir, thanks a lot!

**Comment by levelingup** [comment]
> > @binajmen that worked like a charm
> > tl;dr - create a `my-app.d.ts` and add the below snippet:
> > ```
> > /// <reference types="nativewind/types" />
> > ```
> 
> I named mine app.d.ts and it didn't work but, my-app.d.ts worked or any other name

This also worked for me. any reason why this is the case?

**Comment by Nahsh100** [comment]
> In my case, I've created `app.d.ts` and added `/// <reference types="nativewind/types" />` but I still get typescript error because of className. So, i change the file name `app.d.ts` to `my-app.d.ts` and it works.

This worked for me.

**Comment by zhangela** [comment]
This does not work for me :(

**Comment by zhangela** [comment]
This does not work for me :(

**Comment by MichelleSambiase** [comment]
> No solution here worked for me (Expo w/ typescript mobile app), I finally found out that adding the type in the tsconfig.json works :
> 
> ```
> {
>   "extends": "expo/tsconfig.base",
>   "compilerOptions": {
>     "strict": true,
>     "types": ["nativewind/types"]
>   }
> }
> ```

Thank you so much!!! this approach works for me!

**Comment by alperbayram** [comment]
Create a file "global.d.ts" in the root level of the app and paste this line

`/// <reference types="nativewind/types" />`

This worked for me

**Comment by VictorHCord** [comment]
> > No solution here worked for me (Expo w/ typescript mobile app), I finally found out that adding the type in the tsconfig.json works :
> > ```
> > {
> >   "extends": "expo/tsconfig.base",
> >   "compilerOptions": {
> >     "strict": true,
> >     "types": ["nativewind/types"]
> >   }
> > }
> > ```
> 
> Thank you so much!!! this approach works for me!

**Comment by jimjam88** [comment]
This one was a good old head scratcher... In my pnpm workspace adding "nativewind/types" to my base tsconfig worked fine for my sub-packages but it didn't work for my main expo app.

To get the expo app to work as well, I needed to update the react-native path in the expo directory's tsconfig file:

```json
{
  "extends": "@foo/bar/tsconfig-base.json",
  "compilerOptions": {
    "paths": {
      "react-native": [
        "../../node_modules/react-native",
        "../../node_modules/nativewind"
      ]
    }
  }
}
```

Hints working nicely (including Tailwind). 

Hope this helps somebody!

**Comment by mikespadafora** [comment]
> In my case, I've created `app.d.ts` and added `/// <reference types="nativewind/types" />` but I still get typescript error because of className. So, i change the file name `app.d.ts` to `my-app.d.ts` and it works.

This works for me. Thanks!!!

**Comment by Shreyas-29** [comment]
> In my case, I've created `app.d.ts` and added `/// <reference types="nativewind/types" />` but I still get typescript error because of className. So, i change the file name `app.d.ts` to `my-app.d.ts` and it works.

Yes the error is now gone. But still the styles are not working in my app:
```
import { StatusBar } from 'expo-status-bar';
import { StyleSheet, Text, View } from 'react-native';

export default function App() {
  return (
    <View className='flex-1'>
      <StatusBar style="auto" />
      <Text className="text-red-500">Open up App.tsx to start working on your app!</Text>
    </View>
  );
}
```

**Comment by glennin-codes** [comment]
> Rembember to include app.d.ts to tsconfig.json
> 
> ```
> {
> "include": ["app.d.ts"]
> }
> ```
this indeed is what i forget to do

**Comment by luiz7727** [comment]
> @binajmen that worked like a charm
> 
> tl;dr - create a `my-app.d.ts` and add the below snippet:
> 
> ```
> /// <reference types="nativewind/types" />
> ```

Thanks, it works for me

**Comment by 3flippa** [comment]
> > In my case, I've created `app.d.ts` and added `/// <reference types="nativewind/types" />` but I still get typescript error because of className. So, i change the file name `app.d.ts` to `my-app.d.ts` and it works.
> 
> Yes the error is now gone. But still the styles are not working in my app:
> 
> ```
> import { StatusBar } from 'expo-status-bar';
> import { StyleSheet, Text, View } from 'react-native';
> 
> export default function App() {
>   return (
>     <View className='flex-1'>
>       <StatusBar style="auto" />
>       <Text className="text-red-500">Open up App.tsx to start working on your app!</Text>
>     </View>
>   );
> }
> ```

did you fix it? i have the same problem :(

**Comment by Shreyas-29** [comment]
> > > In my case, I've created `app.d.ts` and added `/// <reference types="nativewind/types" />` but I still get typescript error because of className. So, i change the file name `app.d.ts` to `my-app.d.ts` and it works.
> > 
> > 
> > Yes the error is now gone. But still the styles are not working in my app:
> > ```
> > import { StatusBar } from 'expo-status-bar';
> > import { StyleSheet, Text, View } from 'react-native';
> > 
> > export default function App() {
> >   return (
> >     <View className='flex-1'>
> >       <StatusBar style="auto" />
> >       <Text className="text-red-500">Open up App.tsx to start working on your app!</Text>
> >     </View>
> >   );
> > }
> > ```
> 
> did you fixed it? i have the same problem :(

Try to use this version of tailwindcss:

`npm i tailwindcss@3.3.2`

like this: `"tailwindcss": "3.3.2",`

**Comment by HangMine** [comment]
None of the above will solve my problem

Here's my solution

tsconfig.json
```
{
  "extends": "@react-native/typescript-config/tsconfig.json",
  "include": ["src", "node_modules/nativewind"]
}
```

**Comment by zero19124** [comment]
> If someone is passing by, last recommendation is here: https://www.nativewind.dev/getting-started/typescript

it works thx u

**Comment by saulflores95** [comment]
> This one was a good old head scratcher... In my pnpm workspace adding "nativewind/types" to my base tsconfig worked fine for my sub-packages but it didn't work for my main expo app.
> 
> To get the expo app to work as well, I needed to update the react-native path in the expo directory's tsconfig file:
> 
> ```json
> {
>   "extends": "@foo/bar/tsconfig-base.json",
>   "compilerOptions": {
>     "paths": {
>       "react-native": [
>         "../../node_modules/react-native",
>         "../../node_modules/nativewind"
>       ]
>     }
>   }
> }
> ```
> 
> Hints working nicely (including Tailwind).
> 
> Hope this helps somebody!


This worked, but I still had issues with react-native-paper solved the it using this method

```
    "paths": {
      "~/*": ["./src/*"],
      "react-native": [
        "../../node_modules/react-native",
        "../../node_modules/nativewind"
      ],
      "react-native-paper": [
        "../../node_modules/react-native",
        "../../node_modules/react-native-paper",
        "../../node_modules/nativewind"
      ]
    },
```

**Comment by AmdjedBelgacem** [comment]
None of these solutions worked for me, i mean i have figured it our for Views, but once i reached and added Button i am getting this error :"
No overload matches this call.
  Overload 1 of 2, '(props: ButtonProps): Button', gave the following error.
    Type '{ children: string; onPress: () => void; className: string; title: string; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<Button> & Readonly<ButtonProps>'.
      Property 'children' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<Button> & Readonly<ButtonProps>'.
  Overload 2 of 2, '(props: ButtonProps, context: any): Button', gave the following error.
    Type '{ children: string; onPress: () => void; className: string; title: string; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<Button> & Readonly<ButtonProps>'.
      Property 'children' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<Button> & Readonly<ButtonProps>'."
    My d.ts nativewind file is in the same dir as app, i added the triple slash directive, i included it in the tsconfig, i added nativewind type, i tried the paths trick but all of these don't work, is there something i'm missing ?, Javascript is nice and but i just got used to Typescript

**Comment by saulflores95** [comment]
@osvald0 & @AmdjedBelgacem which nativewind and typescript version are you on?

**Comment by AmdjedBelgacem** [comment]
> @osvald0 & @AmdjedBelgacem which nativewind and typescript version are you on?
I am on: 
"nativewind": "^2.0.11",
"typescript": "~5.3.3"

**Comment by AmdjedBelgacem** [comment]
@saulflores95 I am on :
"typescript": "~5.3.3"
"nativewind": "^2.0.11",

which are fairly new

**Comment by osvald0** [comment]
It's working fine for me now. The versions are:

"nativewind": "^2.0.11",
"typescript": "~5.3.3"

**Comment by saulflores95** [comment]
So we are on the same boat, we all 3 have the same dependencies and none of the solutions above worked.

```
"nativewind": "^2.0.11",
"typescript": "^5.3.3",
"tailwindcss": "3.3.2",
```

**Comment by AmdjedBelgacem** [comment]
@saulflores95 Best Solution for now is to not use Button and replace it with TouchableOpacity

**Comment by Samar-Ali666** [comment]
Just create a file named nativewind.d.ts in root folder with following content

`/// <reference types="nativewind/types" />`

Then in tailwind.config.js do the following

`content: [
    "./App.{js,jsx,ts,tsx}",
    "./<custom directory>/**/*.{js,jsx,ts,tsx}",
  ],`
  
  replace the <custom directory> with your folder name in my case its app
  
  `content: [
    "./App.{js,jsx,ts,tsx}",
    "./app/**/*.{js,jsx,ts,tsx}",
  ],`
  
  It works fine for me

**Comment by milk717** [comment]
> ```json
> -
> ```

In the monolepo structure, this was a very helpful approach.
In my case, it was as follows.

```json
   "react-native": [
        "./node_modules/react-native",
        "../../node_modules/nativewind"
      ]
```

**Comment by milk717** [comment]
> This one was a good old head scratcher... In my pnpm workspace adding "nativewind/types" to my base tsconfig worked fine for my sub-packages but it didn't work for my main expo app.
> 
> To get the expo app to work as well, I needed to update the react-native path in the expo directory's tsconfig file:
> 
> ```json
> {
>   "extends": "@foo/bar/tsconfig-base.json",
>   "compilerOptions": {
>     "paths": {
>       "react-native": [
>         "../../node_modules/react-native",
>         "../../node_modules/nativewind"
>       ]
>     }
>   }
> }
> ```
> 
> Hints working nicely (including Tailwind).
> 
> Hope this helps somebody!



In the monolepo structure, this was a very helpful approach.
In my case, it was as follows.

```json
   "react-native": [
        "./node_modules/react-native",
        "../../node_modules/nativewind"
      ]
```

**Comment by Korin92** [comment]
> No solution here worked for me (Expo w/ typescript mobile app), I finally found out that adding the type in the tsconfig.json works :
> 
> ```
> {
>   "extends": "expo/tsconfig.base",
>   "compilerOptions": {
>     "strict": true,
>     "types": ["nativewind/types"]
>   }
> }
> ```

This is what solved the problem for me. Thanks!!

**Comment by Lipcsyy** [comment]
> No solution here worked for me (Expo w/ typescript mobile app), I finally found out that adding the type in the tsconfig.json works :
> 
> ```
> {
>   "extends": "expo/tsconfig.base",
>   "compilerOptions": {
>     "strict": true,
>     "types": ["nativewind/types"]
>   }
> }
> ```

This just worked perfectly

**Comment by ShedrachJonah11** [comment]
how about in js?

**Comment by khushay25** [comment]
Nothing worked for me nativewind-env.d.ts is getting added automatically, tsconfig metroconfig are rightly configured but still I'm getting this intrinsic property error

**Comment by khushay25** [comment]
Please help

**Comment by AbdeenM** [comment]
@khushay25 if all the comments dont work, try the following:

1. Add the `nativewind-env.d.ts` file containing `/// <reference types="nativewind/types" />` as mentioned in the comments
2. Remove your `node_modules folder(s)`
3. Remove your package lock file i.e. (pnpm-lock.yaml, yarn.lock, etc.)
4. Reinstall your packages i.e. `pnpm install`, `yarn`, etc

Hope that helps!

**Comment by joshualanger2** [comment]
Changing my TS version used in VS Code from 5.7.2 to 4.5.2 resolved this for me in addition to adding the `nativewind-env.d.ts` file containing `/// <reference types="nativewind/types" />` as mentioned in the comments

**Comment by rommyarb** [comment]
Make sure that your `tsconfig.json` file has "nativewind/types" type:
```json
{
  "compilerOptions": {
    "types": [
       ...
       "nativewind/types" // <--- add this
    ],
    ...
  },
  ...
}
```
